lib/actions/usbsdfuTargetActions.js

```diff
@@ -434,6 +434,13 @@ function operateDFU(inputDfuImages, inputDevice) {
                     'The bootloader update must be signed. The DFU operation has been cancelled.'));
             }
 
+            // Notify the user that firmware with gap inside is not supported
+            // The error code refers to pc-nrf-dfu-js
+            if (error.code === ErrorCode.ERROR_TOO_MANY_WRITE_FAILURES) {
```

Comment:
I'm looking at https://github.com/NordicSemiconductor/pc-nrf-dfu-js/blob/8152c7c7765caf04926b5d095cd1e2409ef9cd85/src/DfuAbstractTransport.js#L180 , and that error doesn't happen when the hex files have gaps on them - happens when there have been errors from the DFU target.

Isn't there a better way to determine whether the hex files have gaps in the DFU-able sections? Maybe the DFU target is returning a specific error code, and `pc-nrf-dfu-js` should raise a different (and more specific) error.

Reply:
Let me explain with more detail.
1. If providing a single firmware, with gap, we can do that in practice (for sure, I have done that before), since we know the start and end of the firmware.
2. Now we provide more firmwares at the same time, let's say a softdevice + application. If there is no gap in both softdevice and application, ok, we know the below part is softdevice and the above part is application since there is only one gap between softdevice and application. We cover the case.
3. Then imagine that, there is a gap in softdevice, there will be three parts, softdevice part one, softdevice part two and application. How does the programmer know which is/are belong to softdevice and which is belong t applicaton? (The magic number of softdevice end you mentioned may help, even though it was not exactly correct, I will look into this solution, it is a good tip)
4. And then, how about one gap in softdevice and one gap in application or even more gaps in different firmwares? So the main problem here acually is the approach of detecting the regions, to find out exactly where different firmwares start and end.
5. I remember we discussed this at the first day when I created the definition last year. The conclusion was we cannot cover all the cases. However we cover more and more little by little. 😸 
6. I will find a better error code for it if is possible.